crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -281,39 +355,55 @@ where
-        // or the material uses alpha cutoff values and doesn't rely on the standard prepass shader
-        let fragment = if key.mesh_key.contains(MeshPipelineKey::NORMAL_PREPASS)
-            || ((key.mesh_key.contains(MeshPipelineKey::ALPHA_MASK)
-                || blend_key == MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA)
-                && self.material_fragment_shader.is_some())
-        {
-            // Use the fragment shader from the material if present
-            let frag_shader_handle = if let Some(handle) = &self.material_fragment_shader {
-                handle.clone()
-            } else {
-                PREPASS_SHADER_HANDLE.typed::<Shader>()
-            };
+        // Setup prepass fragment targets - normals in slot 0 (or None if not needed), motion vectors in slot 1
+        let mut targets = vec![];
+        targets.push(
+            key.mesh_key
+                .contains(MeshPipelineKey::NORMAL_PREPASS)
+                .then_some(ColorTargetState {
+                    format: NORMAL_PREPASS_FORMAT,
+                    blend: Some(BlendState::REPLACE),
+                    write_mask: ColorWrites::ALL,
+                }),
+        );
 
-            let mut targets = vec![];
-            // When the normal prepass is enabled we need a target to be able to write to it.
-            if key.mesh_key.contains(MeshPipelineKey::NORMAL_PREPASS) {
-                targets.push(Some(ColorTargetState {
-                    format: TextureFormat::Rgb10a2Unorm,
+        targets.push(
+            key.mesh_key
+                .contains(MeshPipelineKey::MOTION_VECTOR_PREPASS)
+                .then_some(ColorTargetState {
+                    format: MOTION_VECTOR_PREPASS_FORMAT,
                     blend: Some(BlendState::REPLACE),
                     write_mask: ColorWrites::ALL,
-                }));
-            }
+                }),
+        );
+
+        if targets.iter().all(Option::is_none) {
+            // if no targets are required then clear the list, so that no fragment shader is required
+            // (though one may still be used for discarding depth buffer writes)
+            targets.clear();
+        }
+
+        // The fragment shader is only used when the normal prepass or motion vectors prepass
+        // is enabled or the material uses alpha cutoff values
+        let fragment_required = key
+            .mesh_key
+            .intersects(MeshPipelineKey::ALPHA_MASK | MeshPipelineKey::BLEND_PREMULTIPLIED_ALPHA)
+            || !targets.is_empty();
+
+        let fragment = fragment_required.then(|| {
+            // Use the fragment shader from the material
+            let Some(frag_shader_handle) = &self.material_fragment_shader else {
+                // if a fragment shader is required and not provided, we can't continue
+                panic!("No prepass fragment shader provided for {}.", std::any::type_name::<M>());
```

Comment:
```suggestion
                PREPASS_SHADER_HANDLE
````

as discussed on discord, reverting the explicit panic here for now since ShaderRef::Default leads to this branch.